src/main/kotlin/com/demonwav/mcdev/platform/forge/reflection/reference/ReflectedFieldReference.kt

```diff
@@ -0,0 +1,103 @@
+import com.intellij.psi.PsiReferenceProvider
+import com.intellij.psi.ResolveResult
+import com.intellij.psi.codeStyle.CodeStyleManager
+import com.intellij.util.ProcessingContext
+
+object ReflectedFieldReference : PsiReferenceProvider() {
+    override fun getReferencesByElement(element: PsiElement, context: ProcessingContext): Array<PsiReference> {
+        // The pattern for this provider should only match method params, but better be safe
+        if (element.parent !is PsiExpressionList) {
+            return arrayOf()
+        }
+        return arrayOf(Reference(element as PsiLiteral))
+    }
+
+    class Reference(element: PsiLiteral) : PsiReferenceBase.Poly<PsiLiteral>(element) {
+        val fieldName
+            get() = element.constantStringValue ?: ""
+
+        override fun getVariants(): Array<Any> {
+            val typeClass = findReferencedClass() ?: return arrayOf()
+
+            return typeClass.allFields.filter { !it.hasModifier(JvmModifier.PUBLIC) }
+                .map { field ->
+                    JavaLookupElementBuilder.forField(field).withInsertHandler { context, _ ->
+                        val literal = context.file.findElementAt(context.startOffset)?.parent as? PsiLiteral ?: return@withInsertHandler
+
+                        val srgManager = literal.findModule()?.let { MinecraftFacet.getInstance(it) }?.getModuleOfType(McpModuleType)?.srgManager
+                        val srgMap = srgManager?.srgMapNow
+                        val srgField = srgMap?.getSrgField(field.simpleQualifiedMemberReference) ?: return@withInsertHandler
+
+                        context.setLaterRunnable {
+                            // Commit changes made by code completion
+                            context.commitDocument()
+
+                            // Run command to replace PsiElement
+                            CommandProcessor.getInstance().runUndoTransparentAction {
+                                runWriteAction {
+                                    val elementFactory = JavaPsiFacade.getElementFactory(context.project)
+                                    val srgLiteral = elementFactory.createExpressionFromText("\"${srgField.name}\"", literal.parent)
+                                    literal.parent.addBefore(srgLiteral, literal)
+                                    CodeStyleManager.getInstance(context.project).reformat(literal.parent, true)
+                                }
+                            }
+                        }
+                    }
+                }
+                .toTypedArray()
+        }
+
+        override fun multiResolve(incompleteCode: Boolean): Array<ResolveResult> {
+            val typeClass = findReferencedClass() ?: return arrayOf()
+
+            val srgManager = element.findModule()?.let { MinecraftFacet.getInstance(it) }?.getModuleOfType(McpModuleType)?.srgManager
+            val srgMap = srgManager?.srgMapNow
+            val mcpName = srgMap?.mapSrgName(fieldName)
```

Comment:
I'd avoid evaluating `fieldName` twice. Also the branches of the if statement below are really similar, so you can merge them with an additional variable.

Something like:

```kt
val name = fieldName
val mcpName = srgMap?.mapSrgName(name) ?: name
return typeClass.allFields.filter { it.name == name }.map(::PsiElementResolveResult).toTypedArray()
```